bliss/simulator/prior.py

```diff
@@ -148,53 +206,33 @@ def _sample_n_galaxies_and_stars(self, n_sources):
 
         return galaxy_bools, star_bools
 
-    def _sample_star_fluxes(self):
+    def _sample_star_fluxes(self, star_ratios):
         latent_dims = (self.batch_size, self.n_tiles_h, self.n_tiles_w, self.max_sources)
         fluxes = torch.zeros(
             self.batch_size, self.n_tiles_h, self.n_tiles_w, self.max_sources, self.n_bands
         )
-        # draw value for g-band from pareto for each light source
+        # sample value for first band from pareto for each light source
         fluxes[:, :, :, :, 0] = self._draw_truncated_pareto(
             self.star_flux_alpha, self.star_flux_min, self.star_flux_max, latent_dims
         )
-        # Compute r-band via g-band value and ratio from SDSS catalog
-        fluxes[:, :, :, :, 1:] = self.multiflux(6, fluxes[:, :, :, :, 0])
+        # Compute remaining band fluxes via ratio from SDSS catalog and sampled values
+        if self.n_bands > 1:
+            fluxes[:, :, :, :, 1:] = self.multiflux(star_ratios, fluxes[:, :, :, :, 0])
         return fluxes
 
-    def multiflux(self, obj_id, flux_base) -> Tensor:
-        # TODO: Remove fixed camcol/field/run values (or specify in config)
-        camcol = 1
-        field = 12
-        run = 94
-
-        sdss_path = Path(self.sdss)
-        camcol_dir = sdss_path / str(run) / str(camcol) / str(field)
-        po_path = camcol_dir / f"photoObj-{run:06d}-{camcol:d}-{field:04d}.fits"
-        po_fits = fits.getdata(po_path)
-
-        fluxes = column_to_tensor(po_fits, "psfflux")
-        objc_type = column_to_tensor(po_fits, "objc_type").numpy()
-        thing_id = column_to_tensor(po_fits, "thing_id").numpy()
-
-        # TODO: Rewrite so star and galaxy sources are separated from SDSS file
-        # to avoid variable runtime in while loop
-        # pick random light source from SDSS image
-        obj = np.random.randint(0, 999)
-        while objc_type[obj] != obj_id or thing_id[obj] == -1 or torch.any(fluxes[obj] < 0):
-            obj = np.random.randint(0, 999)
-
+    def multiflux(self, rcf_ratios, flux_base) -> Tensor:
```

Comment:
`multiflux` is changed to account for the upstream changes. Now, there is no checking to see whether a sampled light source from a FITS file is a star or a galaxy as this is known beforehand when `multiflux` is called. `multiflux` computes and returns the flux values for all remaining bands scaled according to the draw from the Pareto distribution for the first band specified in the config.